tokenizer_ts/src/tokenizerBuilder.ts

```diff
@@ -0,0 +1,208 @@
+        [ENDOFTEXT, 50256],
+        [FIM_PREFIX, 50281],
+        [FIM_MIDDLE, 50282],
+        [FIM_SUFFIX, 50283]
+      ]);
+      break;
+    default:
+      break;
+  }
+  return specialTokens;
+}
+
+/**
+ * Get the special tokens from the model name
+ * @param modelName model name
+ * @returns Map<string, number> special tokens mapping
+ */
+export function getSpecialTokensByModel(modelName: string): Map<string, number> {
+  let encoderName = getEncoderFromModelName(modelName);
+  let specialTokens: Map<string, number> = getSpecialTokensByEncoder(encoderName);
+  return specialTokens;
+}
+
+/**
+ * Create a tokenizer from a model name
+ * @param modelName model name
+ * @param extraSpecialTokens extra special tokens
+ */
+export async function createByModelName(modelName: string, extraSpecialTokens: ReadonlyMap<string, number> | null = null): Promise<TikTokenizer> {
+  return createByEncoderName(getEncoderFromModelName(modelName), extraSpecialTokens);
+}
+
+/**
+ * Create a tokenizer from an encoder name
+ * @param encoderName encoder name
+ * @param extraSpecialTokens extra special tokens 
+ * @returns TikTokenizer tokenizer
+ */
+export async function createByEncoderName(encoderName: string, extraSpecialTokens: ReadonlyMap<string, number> | null = null): Promise<TikTokenizer> {
+  let regexPattern: string;
+  let mergeableRanksFileUrl: string;
+  let specialTokens: Map<string, number> = getSpecialTokensByEncoder(encoderName);
+
+  switch (encoderName) {
+    case "cl100k_base":
+      regexPattern = REGEX_PATTERN_2;
+      mergeableRanksFileUrl = `https://openaipublic.blob.core.windows.net/encodings/cl100k_base.tiktoken`;
+      break;
+    case "p50k_base":
+      regexPattern = REGEX_PATTERN_1;
+      mergeableRanksFileUrl = `https://openaipublic.blob.core.windows.net/encodings/p50k_base.tiktoken`;
+      break;
+    case "p50k_edit":
+      regexPattern = REGEX_PATTERN_1;
+      mergeableRanksFileUrl = `https://openaipublic.blob.core.windows.net/encodings/p50k_base.tiktoken`;
+      break;
+    case "r50k_base":
+      regexPattern = REGEX_PATTERN_1;
+      mergeableRanksFileUrl = `https://openaipublic.blob.core.windows.net/encodings/r50k_base.tiktoken`;
+      break;
+    case "gpt2":
+      regexPattern = REGEX_PATTERN_1;
+      mergeableRanksFileUrl = `https://pythia.blob.core.windows.net/public/encoding/gpt2.tiktoken`;
+      break;
+    default:
+      throw new Error(`Doesn't support this encoder [${encoderName}]`);
+  }
+
+  if (extraSpecialTokens !== null) {
+    specialTokens = new Map([...specialTokens, ...extraSpecialTokens]);
+  }
+
+  let fileName = getFilenameFromUrl(mergeableRanksFileUrl);
+  await fetch(mergeableRanksFileUrl).then(response => {
+    if (response.ok) {
+      return response.text();
+    } else {
+      throw new Error(`Failed to fetch file from ${mergeableRanksFileUrl}. Status code: ${response.status}`);
+    }
+  })
+    .then(text => {
+      fs.writeFileSync(fileName, text);
```

Comment:
This just creates artifacts in the root directory of the project. At least route them to a `bin/` folder.